packages/grid/x-data-grid-pro/src/tests/columnsVisibility.DataGridPro.test.tsx

```diff
@@ -131,15 +131,19 @@ describe('<DataGridPro /> - Columns Visibility', () => {
     );
 
     fireEvent.click(getByText('Hide all'));
-    expect(getColumnHeadersTextContent()).to.deep.equal([]);
+    await waitFor(() => expect(getColumnHeadersTextContent()).to.deep.equal([]));
     fireEvent.click(document.querySelector('[role="tooltip"] [name="id"]')!);
-    expect(getColumnHeadersTextContent()).to.deep.equal(['id']);
+    await waitFor(() => expect(getColumnHeadersTextContent()).to.deep.equal(['id']));
 
     const separator = document.querySelector(`.${gridClasses['columnSeparator--resizable']}`)!;
-    fireEvent.mouseDown(separator, { clientX: 100 });
-    fireEvent.mouseMove(separator, { clientX: 110, buttons: 1 });
-    fireEvent.mouseUp(separator);
+    act(() => {
```

Comment:
Was it [actually needed](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily)?

Reply:
I don't know why, but it fails with an act warning without it :D

```
1) <DataGridPro /> - Columns Visibility
       should not hide column when resizing a column after hiding it and showing it again:
     Expected test not to call console.error() but instead received 1 calls.

If the warning is expected, test for it explicitly by using the toErrorDev() matcher.
Test location:
  /Users/andrew/workspace/mui-x/packages/grid/x-data-grid-pro/src/tests/columnsVisibility.DataGridPro.test.tsx 

console.error message #1:
  Warning: An update to ForwardRef inside a test was not wrapped in act(...).

When testing, code that causes React state updates should be wrapped into act(...):

act(() => {
  /* fire events that update state */
});
/* assert on the output */

This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
```

Reply:
Used fake timers for it, because we have `setTimeout` in mouseup handler: https://github.com/mui/mui-x/blob/d5cd6702b3344dd568fa89340ce270d05ae3b2ff/packages/grid/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx#LL191C38-L191C45